tests/benchmarks/bench_test.py

```diff
@@ -0,0 +1,60 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+"""Microbenchmarks for CompilerGym environments."""
+import gym
+import pytest
+
+from compiler_gym.envs import CompilerEnv, LlvmEnv
+from compiler_gym.service import CompilerGymServiceConnection
+from tests.test_main import main
+from tests.envs.llvm.fixtures import SERVICE_BIN
+
+pytest_plugins = ["tests.envs.llvm.fixtures"]
+
+# Redefine this fixture since running all of the benchmarks in cBench would
+# take too long, but we do want to use at least one small and one large
+# benchmark to see both per-invocation overhead and overhead that is a result
+# of the size of the fixture.
+#
+# adpcm is small and jpeg-d is large. ghostscript is the largest but that
+# one takes too long.
+@pytest.fixture(params=["cBench-v0/adpcm", "cBench-v0/jpeg-d"])
+def benchmark_name(request) -> str:
+    yield request.param
+
+# The observation benchmark is too slow for a large input.
+@pytest.fixture(params=["cBench-v0/adpcm"])
+def fast_benchmark_name(request) -> str:
+    yield request.param
+
+
+def test_make_local(benchmark):
+    benchmark(lambda: gym.make("llvm-v0").close());
+
+def test_make_service(benchmark):
+    service = CompilerGymServiceConnection(SERVICE_BIN)
+    try:
+      benchmark(lambda: LlvmEnv(service=service.connection.url).close())
+    finally:
+      service.close()
+
+def test_reset(benchmark, env: CompilerEnv, benchmark_name):
+    benchmark(env.reset, benchmark_name)
+
+def test_step(benchmark, env: CompilerEnv, benchmark_name):
+    env.reset(benchmark_name)
+    benchmark(env.step, 0)
+
+def test_observation(benchmark, env: CompilerEnv, fast_benchmark_name):
```

Comment:
Could you add an [observation_space](https://github.com/facebookresearch/CompilerGym/blob/development/tests/envs/llvm/fixtures.py#L49-L51) so that we can benchmark each observation space? They take different amounts of time to compute, it would be great to track those differences.
```suggestion
def test_observation(benchmark, env: CompilerEnv, fast_benchmark_name, observation_space):
```

Reply:
Done.